backend/tracim_backend/views/search_api/schemas.py

```diff
@@ -0,0 +1,245 @@
+
+# from tracim_backend.lib.search.elasticsearch_search.elasticsearch_search import AdvancedContentSearchParameters
+
+
+class SearchContentField(Enum):
+    LABEL = "label"
+    RAW_CONTENT = "raw_content"
+    COMMENT = "comment"
+    DESCRIPTION = "description"
+
+
+class ContentSearchFilterQuery(object):
+    def __init__(
+        self,
+        size: int = 10,
+        page_nb: int = 1,
+        search_string: str = "",
+        content_types: typing.Optional[typing.List[str]] = None,
+        show_deleted: int = 0,
+        show_archived: int = 0,
+        show_active: int = 1,
+    ):
+        self.search_string = search_string
+        self.size = size
+        self.page_nb = page_nb
+
+        if not content_types:
+            self.content_types = content_type_list.restricted_allowed_types_slug()
+
+        self.show_deleted = bool(show_deleted)
+        self.show_archived = bool(show_archived)
+        self.show_active = bool(show_active)
+
+
+class ContentSearchFilterQuerySchema(marshmallow.Schema):
+    search_string = StrippedString(
+        example="test", description="just a search string", required=False
+    )
+    size = marshmallow.fields.Int(
+        required=False, default=10, validate=strictly_positive_int_validator
+    )
+    page_nb = marshmallow.fields.Int(
+        required=False, default=1, validate=strictly_positive_int_validator
+    )
+    content_types = StrippedString(
+        required=False, validate=regex_string_as_list_of_string, description="content_types to show"
+    )
+    show_archived = marshmallow.fields.Int(
+        example=0,
+        default=0,
+        description="if set to 1, then show archived contents."
+        " Default is 0 - hide archived content",
+        validate=bool_as_int_validator,
+    )
+    show_deleted = marshmallow.fields.Int(
+        example=0,
+        default=0,
+        description="if set to 1, then show deleted contents."
+        " Default is 0 - hide deleted content",
+        validate=bool_as_int_validator,
+    )
+    show_active = marshmallow.fields.Int(
+        example=1,
+        default=1,
+        description="if set to 1, then show active contents. "
+        "Default is 1 - show active content."
+        " Note: active content are content "
+        "that is neither archived nor deleted. "
+        "The reason for this parameter to exist is for example "
+        "to allow to show only archived documents",
+        validate=bool_as_int_validator,
+    )
+
+    # @post_load
+    # def make_search_content_filter(self, data: typing.Dict[str, typing.Any]) -> object:
+    # return ContentSearchFilterQuery(
+    # search_string=data["search_string"],
+    # size=data["size"],
+    # page_nb=data["page_nb"],
+    # content_types=data["content_types"],
+    # show_archived=data["show_archived"],
+    # show_deleted=data["show_deleted"],
+    # show_active=data["show_active"],
+    # )
+
+
+class EnumField(marshmallow.fields.Field):
```

Comment:
You should add a docstring to describe what this field does.
By the way you cannot ensure that an enum `.value` will be a string so I suggest to change the `str` to `typing.Any`.
Also this is quite generic so having it with other generic schemas/fields would be useful so that it can be easily re-used.

Reply:
I need the value of the enum to be a string because this is what is used for serialization. We could use str(.value) though.

Reply:
okay, on second look It's better indeed not to be too restrictive here.